### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by JeanGodGandanne**

### CODE REVIEW

File: database/src/main/java/invivible/database/models/objects/Rating.java [hunk 1]
```diff
@@ -32,7 +33,11 @@
 
   @Id
   private Long id;
+  private Long userId;
+  @Transient
   private User user;
+  private Long poiId;
+  @Transient
   private PointOfInterest poi;
```
Comment (simonbinder) [inline database/src/main/java/invivible/database/models/objects/Rating.java]: warum haben wir hier nochmal den Poi, wenn in der Entity auch die poiId drin ist?

**Comment by simonbinder** [comment]
Ich habs gerade nochmal getestet und bekomme immer noch die Nullpointer Exception, bin mir aber nicht sicher ob das nicht vielleicht auch an meinen Daten liegt. Hier ist das JSON-Objekt, das ich gepostet hab: 
`{
	"categorieRatings": [
		{
			"comment": "",
			"id": null,
			"rating": "FALSE",
			"tag": [],
			"question": {
				"id": 1,
				"text": "Does the institution provide social conduct guidelines for prevention of and/or dealing with racism?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": [
					4
				]
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 2,
				"text": "Does the institution provide social conduct guidelines for prevention of and/or dealing with transgender, intersex and non binary inclusion?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": [
					4
				]
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 3,
				"text": "Does the institution provide social conduct guidelines for prevention of and/or dealing with sexualised violence?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": [
					4
				]
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 4,
				"text": "Please enter the date of the policy and maybe provide a link to the policy, if you can.",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": false,
				"followUpQuestion": null
			}
		},
		{
			"comment": "Ein Kommentar",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 5,
				"text": "Are the existing policies for anti-discrimination/violence, transparent, public and accessible?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": [
					6
				]
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "FALSE",
			"tag": [],
			"question": {
				"id": 6,
				"text": "Are the social conduct guidelines/ policies available in English?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [
				"test"
			],
			"question": {
				"id": 7,
				"text": "Is there a designated person at the institution who you can talk to in case of discrimination?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 8,
				"text": "Is there a designated person at the institution who you can talk to in case of sexualised violence?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 9,
				"text": "Is there a designated person at the institution who you can talk to in case of discrimination against you as a TIN-person?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 10,
				"text": "Is there a designated person at the institution who you can talk to in case of discrimination against you as a person of colour?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 11,
				"text": "Is there a designated person at the institution who you can talk to in case of discrimination covered by the AGG?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 12,
				"text": "Is there a designated person at the institution who you can talk to in case of you as a person with a physical disability?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 13,
				"text": "Is there a designated person at the institution who you can talk to in case of you as a person with a chronic disease?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 14,
				"text": "Is there someone at the institution who is generally responsible for diversity?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 15,
				"text": "Is there someone at the institution who is trained in the resolution of intersectional conflicts?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 16,
				"text": "Does the institution provide trainings and qualification for administrative and counselling staff, students and professors concerning TIN-respectful interaction?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 17,
				"text": "Do you identify as a TIN person?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": [
					18
				]
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 18,
				"text": "Do you feel safe as TIN person at the institution?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 19,
				"text": "Are TIN-people included in social groups such as athletic programs, cultural and political groups?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 20,
				"text": "Does the institution provide service facilities specifically addressing TIN university members (e.g. study counselling, mental healthcare, psychological counselling)?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 21,
				"text": "Do you identify as a TIN person of color? E.g do you identify as racialized, migrant, asylum / refugee, foreign student, other?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": [
					22
				]
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 22,
				"text": "Do you feel safe as person of color at the institution?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 23,
				"text": "Does the institution provide trainings and qualification for administrative and counselling staff, students and professors concerning respectful interaction with POC (people of color)?",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 24,
				"text": "If you have experienced violence or discrimination, and wish to share your story add your story (it will be anonymized). If possible please specify the date (month/year) when you experienced this.",
				"category": {
					"id": 1,
					"name": "Overview: Guidelines and Diversity Management"
				},
				"subCategory": {
					"id": null,
					"name": null,
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 25,
				"text": "Does the institution provide gender neutral toilets?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 1,
					"name": "Architecture and Building Design",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 26,
				"text": "Does the institution provide single stall toilets open to all genders?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 1,
					"name": "Architecture and Building Design",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 27,
				"text": "Does the institution provide single stall showers open to all genders?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 1,
					"name": "Architecture and Building Design",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 28,
				"text": "Does the institution provide single stall changing rooms open to all genders?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 1,
					"name": "Architecture and Building Design",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 29,
				"text": "Are toilets, showers, changing rooms designed in a way that relies on the function of the room and not the sex characteristics of a person?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 1,
					"name": "Architecture and Building Design",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 30,
				"text": "Are the forms and questionnaires TIN-inclusively designed?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 2,
					"name": "Organization of work, study and teaching",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 31,
				"text": "Do you consider the IT-infrastructure adequate for TIN people in the creation of user accounts and the creation and maintenance of personal files?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 2,
					"name": "Organization of work, study and teaching",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 32,
				"text": "Do regulations exist which enable university members to the change their first names, surnames and gender marker on campus records without changing legal documents?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 2,
					"name": "Organization of work, study and teaching",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": [
					33
				]
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 33,
				"text": "Please specify which documents are required and provide a link to the regulation, if you can.",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 2,
					"name": "Organization of work, study and teaching",
					"category": null
				},
				"hasCheckbox": false,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 34,
				"text": "Do procedures exist for university members to provide information about their preferred salutation and pronoun?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 2,
					"name": "Organization of work, study and teaching",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": [
					35
				]
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 35,
				"text": "Please specify which documents are required and provide a link to the regulation as well, if you can.",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 2,
					"name": "Organization of work, study and teaching",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 36,
				"text": "Do ethical TIN-related teaching standards exist in your faculty?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 3,
					"name": "Contents of researching and teaching",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 37,
				"text": "How well are ethical TIN-related teaching standards applied in your faculty?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 3,
					"name": "Contents of researching and teaching",
					"category": null
				},
				"hasCheckbox": false,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 38,
				"text": "Do ethical TIN-related research standards exist in your faculty?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 3,
					"name": "Contents of researching and teaching",
					"category": null
				},
				"hasCheckbox": true,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 39,
				"text": "How well are ethical TIN-related research standards applied in your faculty?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 3,
					"name": "Contents of researching and teaching",
					"category": null
				},
				"hasCheckbox": false,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 40,
				"text": "How well is the history of the university and it's involvement in the oppression of TIN people scientifically analysed?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 3,
					"name": "Contents of researching and teaching",
					"category": null
				},
				"hasCheckbox": false,
				"followUpQuestion": null
			}
		},
		{
			"comment": "",
			"id": null,
			"rating": "YES",
			"tag": [],
			"question": {
				"id": 41,
				"text": "How well is the history of the university and it's involvement in the oppression of people of color scientifically analysed?",
				"category": {
					"id": 2,
					"name": "Monitoring of specific policy aspects"
				},
				"subCategory": {
					"id": 3,
					"name": "Contents of researching and teaching",
					"category": null
				},
				"hasCheckbox": false,
				"followUpQuestion": null
			}
		}
	],
	"generalComment": "Tolle Hochschule!",
	"id": null,
	"userId": 8,
	"poiId": 76738
}`

File: database/src/main/java/invivible/database/models/objects/Rating.java (hunk 1 above)
Comment (JeanGodGandanne) [reply database/src/main/java/invivible/database/models/objects/Rating.java]: Ja brauchen wir vllt gar nicht.

**Comment by JeanGodGandanne** [comment]
Ich gucks mir nachher mal an

**Comment by JeanGodGandanne** [comment]
@simonbinder habs ein wenig angepasst (hoffentlich vereinfacht). Rating save funkltioniert bei mir, wenn das bei dir immernoch nciht klappt müssen wir da mal genauer drauf gucken.

Dein Rating obejkt sollte jetzt also irgendwie so aussehen:

{ 
"categorieRatings": 
	[ 
		{ "comment": "some comment", 
		  "rating": "FALSE", 
		  "tag": ["FB4"], 
		  "questionId": 1
		},
// plus alle weiteren fragen die du beantwortet hast
], 
"generalComment": "Tolle Hochschule!", 
"userId": 5, 
"poiId": 76738 
}

**Comment by simonbinder** [comment]
@JeanGodGandanne bei mir sehen jetzt alle Fragen so aus, wenn ich die aus dem Backend hole: `  {
      "id": 1,
      "text": "Does the institution provide social conduct guidelines for prevention of and/or dealing with racism?",
      "categoryId": null,
      "subCategoryId": null,
      "hasCheckbox": true,
      "followUpQuestion": [
        4
      ]`
Sprich categoryId und subCategoryId sind null. Brauch ich dafür einen neuen DB-Stand?

**Comment by JeanGodGandanne** [comment]
@simonbinder Ja ist korrekt. Der jetzige Stand hat noch die fat objects

**Comment by simonbinder** [comment]
@JeanGodGandanne ich merge jetzt auf jeden Fall mal, sieht gut aus

**Pull request merged**